server/src/main/java/org/opensearch/index/codec/composite/composite912/Composite912DocValuesWriter.java

```diff
@@ -277,7 +278,8 @@ public SortedNumericDocValues getSortedNumeric(FieldInfo field) {
     }
 
     private boolean isSortedSetField(String field) {
-        return mapperService.fieldType(field) instanceof KeywordFieldMapper.KeywordFieldType;
+        return mapperService.fieldType(field) instanceof KeywordFieldMapper.KeywordFieldType
+            || mapperService.fieldType(field) instanceof IpFieldMapper.IpFieldType;
```

Comment:
how can we support sorted set fields from plugins with this approach?

Reply:
https://github.com/opensearch-project/OpenSearch/issues/16642 - yeah we have an open issue for this. 

I'm thinking if we can use fieldDataBuilder of the type as below.

```
 private boolean isSortedSetField(String field) {
        IndexFieldData.Builder builder = mapperService.fieldType(field).fielddataBuilder(
            "index",
            () -> { throw new UnsupportedOperationException();
            });

        if (builder instanceof SortedSetOrdinalsIndexFieldData.Builder) {
            return true;
        }
        return false;
    }
```
Let me know your thoughts.

Reply:
`IndexNumericFieldData` / `IndexOrdinalsFieldData` too we can use. [ they're the parent base classes for ordinals / numerics ]

Reply:
Would this be enough since we need to define DimensionType as well (and maybe more??)?
Lets open an issue to discuss this but we need to add support for allowing field types without requiring changes in core.

Reply:
As discussed offline, I tried extending the builder to rather return a trifunction of dimension. But there is a flow in mapper [ post flattening] which still was dependent on core types [ there is no context of builder]. 

I also looked into other fields such as `derived` fields and even they're backed with a type in core.

With the support of keyword/ordinal and numeric dimensions, hopefully the plugins are able to fit in. 

Let me know what you think.

Reply:
Creating an issue to make pluggable interfaces for star tree [ mainly dimensions ]